### ISSUE TITLE
job starts failed driver [exec]

### ISSUE THREAD

**Issue by FlankMiao** [hashicorp/nomad#5387]
### Nomad version
Output from `nomad version`
0.9.0-beta3

### Operating system and Environment details
OS: centos
OS version: 7.6.1810
### Issue
driver [exec]:
the exec job has 11 arguments.

error log in the nomad-ui:
standard_init_linux.go:207: exec user process caused "argument list too long"
2019-03-05T02:57:47.250Z [ERROR] failed to initialize libcontainer-shim: error="standard_init_linux.go:207: exec user process caused "argument list too long""
panic: --this line should have never been executed, congratulations--

goroutine 1 [running, locked to thread]:
main.init.1()
	/home/mygo/src/github.com/hashicorp/nomad/nsenter_linux.go:23 +0x1d1

log in nomad:
2019-03-04T21:59:18.070-0500 [ERROR] client.driver_mgr.exec: error receiving stream from Stats executor RPC, closing stream: alloc_id=e9163326-ce8e-f42a-2d26-90377d418415 driver=exec task_name=mars error="rpc error: code = Unavailable desc = transport is closing"
2019-03-04T21:59:18.072-0500 [INFO ] client.alloc_runner.task_runner: restarting task: alloc_id=e9163326-ce8e-f42a-2d26-90377d418415 task=mars reason="Restart within policy" delay=26.929660818s
2019-03-04T21:59:45.034-0500 [INFO ] client.driver_mgr.exec: starting task: driver=exec driver_cfg="{Command:mars Args:[...]}"
2019-03-04T21:59:45.582-0500 [ERROR] client.alloc_runner.task_runner: running driver failed: alloc_id=e9163326-ce8e-f42a-2d26-90377d418415 task=mars error="failed to launch command with executor: rpc error: code = Unknown desc = container_linux.go:344: starting container process caused "process_linux.go:424: container init caused \"\"""
2019-03-04T21:59:45.602-0500 [INFO ] client.alloc_runner.task_runner: not restarting task: alloc_id=e9163326-ce8e-f42a-2d26-90377d418415 task=mars reason="Error was unrecoverable"
2019-03-04T21:59:45.648-0500 [INFO ] client.gc: marking allocation for GC: alloc_id=e9163326-ce8e-f42a-2d26-90377d418415

after 2~3 times task restart, it starts correctly.

**Comment by preetapan** [comment]
@FlankMiao Thanks for reporting this, we will fix this before 0.9 final.

**Comment by notnoop** [comment]
@FlankMiao The unexpected panic has been fixed in https://github.com/hashicorp/nomad/pull/5415 .

However, we were unable to reproduce a case resulting in 
```
driver [exec]:
the exec job has 11 arguments.

error log in the nomad-ui:
standard_init_linux.go:207: exec user process caused "argument list too long"
```

It's also especially odd that the task eventually starts, considering the number of arguments doesn't change between starts.  Mind in you provide the full job spec, along with `ulimit -a` output in our system?
